src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/GetAzureRmDataLakeStoreItemContent.cs

```diff
@@ -16,86 +16,133 @@
-            ConfirmAction(
-                Resources.DownloadFileDataMessage,
-                Path.TransformedPath,
-                () =>
-                {
-                    byte[] byteArray;
-                    if (Length <= 0)
+            if (this.ParameterSetName.Equals(BaseParameterSetName, StringComparison.OrdinalIgnoreCase))
+            {
+                ConfirmAction(
+                    Resources.DownloadFileDataMessage,
+                    Path.TransformedPath,
+                    () =>
                     {
-                        Length = (long)DataLakeStoreFileSystemClient.GetFileStatus(Path.TransformedPath, Account).Length - Offset;
-                        if (Length > 1 * 1024 * 1024 && !Force)
+                        byte[] byteArray;
+                        if (Length <= 0)
+                        {
+                            Length = (long)DataLakeStoreFileSystemClient.GetFileStatus(Path.TransformedPath, Account).Length - Offset;
+                            if (Length > 1 * 1024 * 1024 && !Force)
                         // If content is greater than 1MB throw an error to the user to let them know they must pass in a length to preview this much content
                         {
-                            throw new InvalidOperationException(string.Format(Resources.FilePreviewTooLarge, 1 * 1024 * 1024, Length));
+                                throw new InvalidOperationException(string.Format(Resources.FilePreviewTooLarge, 1 * 1024 * 1024, Length));
+                            }
                         }
-                    }
 
-                    using (var memStream = ((MemoryStream)DataLakeStoreFileSystemClient.PreviewFile(Path.TransformedPath, Account, Length, Offset,
-                        CmdletCancellationToken, this)))
-                    {
-                        byteArray = memStream.ToArray();
-                    }
+                        using (var memStream = ((MemoryStream)DataLakeStoreFileSystemClient.PreviewFile(Path.TransformedPath, Account, Length, Offset,
+                            CmdletCancellationToken, this)))
+                        {
+                            byteArray = memStream.ToArray();
+                        }
 
-                    if (UsingByteEncoding(Encoding))
-                    {
-                        WriteObject(byteArray);
-                    }
-                    else
-                    {
-                        WriteObject(BytesToString(byteArray, Encoding));
-                    }
-                });
+                        if (UsingByteEncoding(Encoding))
+                        {
+                            WriteObject(byteArray);
+                        }
+                        else
+                        {
+                            WriteObject(BytesToString(byteArray, Encoding));
+                        }
+                    });
+            }
+            else
+            {
+                if(Head > 0 && Tail > 0)
```

Comment:
@begoldsm based on the conditions you have below, I think it'd be best if you split `Head` and `Tail` into their own parameter sets, which will guarantee they can't be used together. Then you can check which parameter (or parameter set) is being used and not have to worry about checking for the case(s) they are used together. Thoughts?